### PR TITLE
Revised attribution

### PR DESCRIPTION
**Pull request by tristen**
Hits https://github.com/mapbox/mapbox.js/issues/744
- Attribution control is now default over of infoControl.
- infoControl is now triggered when it's set to true in the `L.mapbox.map` options param or when passed as an argument to `map.addControl()`.
- **Improve this map** link now tracks on the attribution control
### TODO
- [x] Mark change in CHANGELOG
- [x] Test coverage for **Improve this map** in L.control.attribution.

cc/ @tmcw for a review.

### CODE REVIEW

File: src/map.js [hunk 1]
```diff
@@ -139,11 +155,15 @@ var LMap = L.Map.extend({
             this.attributionControl.addAttribution(layer.getAttribution());
         }
 
+        this.off('moveend', this._editLink, this);
```
Comment (tmcw) [inline src/map.js]: What is this doing - why unbind and rebind the same function to the same event?

File: test/spec/info_control.js [hunk 2]
```diff
@@ -93,7 +95,9 @@ describe('L.mapbox.infoControl', function() {
     it('adds map location to improve link in attribution when .mapbox-improve-map is present', function(done) {
         var server = sinon.fakeServer.create();
         var element = document.createElement('div');
-        var map = L.mapbox.map(element, 'examples.h8e9h88l');
```
Comment (tmcw) [inline test/spec/info_control.js:96]: Doesn't look like any test coverage is added for the ability to click 'improve this map' in L.control.attribution, not just infoControl

**Comment by tmcw** [comment]
Made inline comments for potentially tricky sections of code. Besides that, this'll need an update in the CHANGELOG.md.

File: src/map.js [hunk 3]
```diff
@@ -128,6 +127,23 @@ var LMap = L.Map.extend({
         }
     },
 
+    _editLink: function() {
+        if (!this._controlContainer.getElementsByClassName) return;
+        var link = this._controlContainer.getElementsByClassName('mapbox-improve-map');
+        if (link.length && this._loaded) {
+            var center = this.getCenter().wrap();
+            var tilejson = this._tilejson || this._map._tilejson || {};
```
Comment (tmcw) [inline src/map.js]: `this._map._tilejson` will never mean anything here.

**Comment by tmcw** [comment]
Okay: my only remaining concern is that the visual difference here will be quite significant for long attribution strings:

2014-05-14 at 2 03 pm

These are, of course, the minority, and might be fine as-is. Thoughts @tristen + @lxbarth ?

**Comment by tristen** [comment]
That seems like a infoControl instead of L.control.attribution use case.

**Comment by tmcw** [comment]
Yes, but the concern is that by switching these on Mapbox.com, we'll turn an (i) into the above - there's no plan of making this a user-facing UI option (do you want less attribution or more?), so should we be fine with long long strings, or should we do something more auto?

**Comment by tristen** [comment]
My heart is telling me to merge for now. The auto solution could be good if this turns into a bad problem.

**Pull request merged**